extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/QuarkusErrorServlet.java

```diff
@@ -40,15 +40,17 @@ protected void service(HttpServletRequest req, HttpServletResponse resp) throws
         if (accept != null && accept.contains("application/json")) {
             resp.setContentType("application/json");
             resp.setCharacterEncoding(StandardCharsets.UTF_8.name());
-            String escapedStack = stack.replace(System.lineSeparator(), "\\n").replace("\"", "\\\"");
-            StringBuilder jsonPayload = new StringBuilder("{\"details\":\"").append(details).append("\",\"stack\":\"")
-                    .append(escapedStack).append("\"}");
+            String escapedDetails = escapeJsonString(details);
+            String escapedStack = escapeJsonString(stack);
+            StringBuilder jsonPayload = new StringBuilder("{\"details\":\"").append(escapedDetails)
+                    .append("\",\"stack\":\"").append(escapedStack).append("\"}");
             resp.getWriter().write(jsonPayload.toString());
         } else {
             //We default to HTML representation
+            String escapedDetails = escapeHtml(details);
```

Comment:
I think it is escaped later in `TemplateHtmlBuilder` so we don't need to escape it here. Am I right?

Reply:
Yes, it is also escaped in the `TemplateHtmlBuilder` directly in the used constructor. I just removed the escaping from QuarkusErrorServlet.